chromium_src/chrome/browser/ui/webui/settings/settings_secure_dns_handler.cc

```diff
@@ -0,0 +1,50 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "base/values.h"
+#include "brave/components/unstoppable_domains/buildflags/buildflags.h"
+
+#if BUILDFLAG(UNSTOPPABLE_DOMAINS_ENABLED)
+#include "brave/net/unstoppable_domains/constants.h"
+#endif
+
+namespace settings {
+namespace {
+std::unique_ptr<base::DictionaryValue> CreateSecureDnsSettingDict();
+}  // namespace
+}  // namespace settings
+
+#include "../../../../../../../chrome/browser/ui/webui/settings/settings_secure_dns_handler.cc"
+
+namespace settings {
+
+namespace {
+
+// Hide Unstoppable Domains resolver in the custom provider list in settings
+// because it will be used for name resolution only for TLDs from Unstoppable
+// Domains, instead of a global DoH setting.
+std::unique_ptr<base::DictionaryValue> CreateSecureDnsSettingDict() {
```

Comment:
This is probably a more stable patch, but wouldn't it be easier to just insert a patch into the loop that appends values to `secure_dns_templates`?

Reply:
@mkarolin I was trying to make it a more stable and maybe flexible patch to just patch the original function name by appending _ChromiumImpl and have our own function here.
I can definitely change to just patch the loop itself if preferred.

Reply:
Changed in eda54f0